csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -679,6 +644,60 @@ private void VerifyNativeMethodsExist()
             }
         }
 
+        private bool RunInferenceOnAModel(uint graphOotimizationLevel)
```

Comment:
Can we keep the Original **CanRunInferenceOnAModel**() unit test as it was (rename it to **RunInferenceOnAModel**)? That way it can double as a user sample as well (i.e. keep a simple version).

We can add another Fact called **RunInferenceOnAModelWithSessionOptions**, which shows how to set the graph optimization level (and we can drop the metaData checks). I.e. just have a simple version which showcases/tests graph optimization.

Reply:
With this change we do retain the original version of the test... I am just moving some code around to avoid code duplication.... Basically all the common code is now in RunInferenceOnAModel helper functions.... when called from Can RunInference all optimizations are disabled and therefore it is the same as before... 

I do want to add a test which runs inference with all options enabled and with the metadata checks... without these checks we cannot guarantee there was no issue  

Regarding a sample version for user - we already have InferenceSample

Reply:
In that case, we can convert the [Fact] into a [Theory] instead, and pass in the optimization levels accordingly. No need to set up 2 fact functions, and 1 non-fact function (see below)

        [Theory]
        [InlineData(0)]
        [InlineData(2)]
        private void RunInferenceOnAModel(uint optimizationLevel)
        {
        .......
        }